PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,33 @@
+## Why?
+
+( Please include a rationale for the changes and the related issue, providing any relevant motivation and context. )
+
+( Please include a link to the Linear ticket(s). If there is no ticket associated, explain why. )
+( e.g. [ARCH-66 : chore: Create a standard PR Template](https://linear.app/archipelago-ai/issue/ARCH-66/chore-create-a-standard-pr-template) )
+
+## This PR
+
+( Please summarize the changes in this PR, using plain English wherever possible. )
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## Screenshots (if applicable)
+
+## Testing
+
+( Please describe the tests that you ran to verify your changes. Consider unit, integration, and functional (manual) testing. Ideally, provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Not all changes can be or need to be tested. If so, please write "N/A because <reason>" )
+
+
+## Checklist:
+
+- [ ] I have performed a self-review of my own code
+- [ ] I have made corresponding changes to the documentation
+- [ ] I have merged the latest changes from `main`
+- [ ] CI is Green ✅ 
```

Comment:
The last two seem redundant. If there is conflict or failed build, the merge will fail.

Reply:
It seems that way but there are some occasions where mainline is green, the branch is green, the merge to main has no conflicts but results in a failed build (or successful build with runtime bugs). 

There do exist GHA checks to build the branch after merged with master, they were used at my last gig and worked pretty well to catch these edge-cases. but I don't know off-hand how to set them up. We can certainly look into it. The manual checklist will help us build good habits.

We can also combine the last 2 somehow to keep it really slim, I'm not opposed. But there will be that small risk factor. 🤷

Reply:
Remember that we run CI not just for PR commits but also for the merge, though the `push` event, so we should catch most problems. See also [Running your pull_request workflow when a pull request merges](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges)